packages/vite-node/src/client.ts

```diff
@@ -59,6 +59,9 @@ export class ViteNodeRunner {
         const resolvedDep = await this.options.resolveId(dep, id)
         dep = resolvedDep?.id?.replace(this.root, '') || dep
       }
+      if (!isAbsolute(dep) && extname(dep))
+        dep = normalize(`${dirname(id)}/${dep}`)
```

Comment:
Are you sure this is not an error in the `if` above?

Reply:
i.e. my exact dynamic import is `import('../../Component.vue')`

Reply:
I mean that `dep` can be wrongly resolved in an `if` statement before - I see you use `!isAbsolute`, this is the same condition for it. Did you try removing it and see if it helps in your case?

If so, we should try fixing it instead of creating another patch on top of it 👀

Reply:
Also I guess it isn't absolute path anymore since I forgot I am normalizing to id instead of filePath. Id is just the path to the file doing the import from the current working directory. Updating PR.

Reply:
Also can you create a test case to cover this?

Reply:
Also I guess it isn't absolute path anymore since I forgot I am normalizing to id instead of filePath. Id is just the path to the file doing the import from the current working directory. Updating PR name.

Reply:
> Also I guess it isn't absolute path anymore since I forgot I am normalizing to id instead of filePath. Id is just the path to the file doing the import from the current working directory. Updating PR.

I would suggest creating a reproduction in `tests`, so others can help you

Reply:
I saw you had tests in the examples directory for this, but I don't think they are exactly ran? Or are there tests for this situation. I was looking over your changes for handling dynamic imports that have variables.

Reply:
hmmmm  yea I  guess you are right on that..... resolvedDep was null so it would fall back to dep. I guess I can move the check inside before the resolveId call and just see if it has an extname.

Reply:
> I saw you had tests in the examples directory for this, but I don't think they are exactly ran? Or are there tests for this situation. I was looking over your changes for handling dynamic imports that have variables.

Create a test that fails without your changes.

Reply:
> hmmmm yea I guess you are right on that..... resolvedDep was null so it would fall back to dep. I guess I can move the check inside before the resolveId call and just see if it has an extname.

not sure if this is the right solution. `resolveId` should always return path if it exists, sometrhing seems fishy